tasks/agent-macos.yml

```diff
@@ -0,0 +1,175 @@
+---
+- name: 'Load user data into JSON string {"uid": <number>, "gid": <number>}'
+  shell:
+    cmd: "dscacheutil -q user -a name {{ ansible_user }} | awk 'BEGIN { RS=\"\\n\"; ORS=\" \" } /uid:/ { print \"{ \\\"uid\\\": \" $2\",\" } /gid:/ { print \"\\\"gid\\\": \" $2 \" }\"}'"
+    executable: /bin/bash
+  changed_when: false
+  register: macos_user_output
+
+- name: Extract JSON user data as variable object
+  set_fact:
+    macos_user_data: "{{ macos_user_output.stdout }}"
+
+- name: Check if system plist file already exists
+  stat:
+    path: "{{ datadog_macos_system_plist_file_path }}"
+  register: macos_system_plist_file_stat
+
+- name: Execute Block when system plist file did not already exist
+  block:
+  - name: Move plist file to system location
+    command:
+      cmd: "mv /Users/{{ ansible_user }}/{{ datadog_macos_user_plist_file_path }} {{ datadog_macos_system_plist_file_path }}"
+      removes: "/Users/{{ ansible_user }}/{{ datadog_macos_user_plist_file_path }}"
+      creates: "{{ datadog_macos_system_plist_file_path }}"
+    become: true
+
+  - name: Inject user and group data into system plist file
+    shell:
+      cmd: "cp -f {{ datadog_macos_system_plist_file_path }} /tmp/.plist.backup
+            && 
+            awk -v d=\"</dict>\"
```

Comment:
i chose to use `awk` because at Applause we often install `gnu-sed` over the built in `sed` (BSD) and they are a bit different.

Reply:
Sounds like a good choice 👍